vllm/core/scheduler.py

```diff
@@ -1058,13 +1061,16 @@ def free_seq(self, seq: Sequence) -> None:
         self.block_manager.free(seq)
 
     def free_finished_seq_groups(self) -> None:
-        for queue in [self.running, self.swapped, self.waiting]:
```

Comment:
I notice that this code is introduced by https://github.com/vllm-project/vllm/pull/4115 @mzusman . Please take a look to see if it makes sense to you.

Reply:
Yeah, Looks really good. We've missed that use-case. Thank you for fixing it up!